src/main/scala/beam/agentsim/agents/modalBehaviors/DrivingMode.scala

```diff
@@ -0,0 +1,132 @@
+package beam.agentsim.agents.modalBehaviors
+
+import akka.actor.ActorRef
+import beam.agentsim.agents.PersonAgent.{PersonData, ScheduleBeginLegTrigger}
+import beam.agentsim.agents.modalBehaviors.DrivingMode.CompleteLegTrigger
+import beam.agentsim.agents.modalBehaviors.DrivingMode._
+import beam.agentsim.agents.util.{AggregatorFactory, MultipleAggregationResult}
+import beam.agentsim.agents.vehicles.BeamVehicle._
+import beam.agentsim.agents.vehicles.VehicleData._
+import beam.agentsim.agents.vehicles.{EnterVehicleTrigger, LeaveVehicleTrigger, VehicleData}
+import beam.agentsim.agents.{BeamAgent, PersonAgent, TriggerShortcuts}
+import beam.agentsim.events.resources.vehicle._
+import beam.agentsim.scheduler.{Trigger, TriggerWithId}
+import beam.router.RoutingModel.BeamLeg
+import beam.sim.HasServices
+
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+
+/**
+  * @author dserdiuk on 7/29/17.
+  */
+object DrivingMode {
+  case class CompleteLegTrigger(tick: Double, beamLeg: BeamLeg) extends Trigger
+  implicit val beamLegOrdering: Ordering[BeamLeg] = Ordering.by(_.startTime)
+
+}
+
+trait DrivingMode extends BeamAgent[PersonData] with TriggerShortcuts with HasServices with AggregatorFactory {
+  this: PersonAgent =>
```

Comment:
If possible I'd like this to be independent of PersonAgent, we will want non-persons to be able to drive. E.g. for now our transit drivers won't be persons and we will likely want to create AI drivers in the future as well.